acados/sim/sim_gnsf.h

```diff
@@ -0,0 +1,253 @@
+/*
+ *    This file is part of acados.
+ *
+ *    acados is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation; either
+ *    version 3 of the License, or (at your option) any later version.
+ *
+ *    acados is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with acados; if not, write to the Free Software Foundation,
+ *    Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * 
+ *    Author: Jonathan Frey
+ */
+
+#ifndef ACADOS_SIM_SIM_GNSF_H_
+#define ACADOS_SIM_SIM_GNSF_H_
+
+#include <stdbool.h>
+
+#include "acados/utils/timing.h"
+#include "acados/utils/types.h"
+
+#include "acados/sim/sim_common.h"
+
+#include "blasfeo/include/blasfeo_target.h"
+#include "blasfeo/include/blasfeo_common.h"
+#include "blasfeo/include/blasfeo_d_aux_ext_dep.h"
+#include "blasfeo/include/blasfeo_d_blas.h"
+#include "blasfeo/include/blasfeo_d_kernel.h"
+#include "blasfeo/include/blasfeo_i_aux_ext_dep.h"
+#include "blasfeo/include/blasfeo_d_aux.h"
+
+typedef struct
+{
+    int nx;
+    int nu;
+    int num_stages;
```

Comment:
Once the model contains only external functions and A, B, ..., L_x, ... and not pre-computed quantities, there is no need to have num_stages and num_steps in the dimensions, but they should be read from opts (same as for the other integrators). Right?